File: java/ClosestPoints.java
class Calculate
{
	public static double Distance(int x, int y)
	{
		return x * x + y * y;
	}
}

class Solution
{
	public static void Sort(int[][] array)
	{
		int minIndex = 0;

		for(int i = 0; i < array.length; i++)
		{
			minIndex = i;

			for(int j = i + 1; j < array.length; j++)
			{
				if(Calculate.Distance(array[j][0], array[j][1]) < Calculate.Distance(array[minIndex][0], array[minIndex][1])) minIndex = j;
			}

			int a0 = array[i][0];
			int a1 = array[i][1];
			int b0 = array[minIndex][0];
			int b1 = array[minIndex][1];

			array[i][0] = b0;
			array[i][1] = b1;
			array[minIndex][0] = a0;
			array[minIndex][1] = a1;
		}
	}

	public static int[][] ClosestPoints(int[][] points, int k)
	{
		if(points[0].length == 0) return points;
		int[][] closest = new int[k][];

		Sort(points);

		for(int i = 0; i < k; i++)
		{
			closest[i] = new int[2];
			closest[i][0] = points[i][0];
			closest[i][1] = points[i][1];
		}

		return closest;
	}
}

public class Medium_K_Closest_Points
{
	public static void main(String[] args)
	{
		int k = 4;
		int size = 10;
		int[][] points = new int[size][];

		for(int i = 0; i < size; i++)
		{
			points[i] = new int[2];
			points[i][0] = size - i - 1;
			points[i][1] = size - i - 1;
		}

		int[][] closest = Solution.ClosestPoints(points, k);

		for(int i = 0; i < k; i++)
		{
			System.out.println("[" + closest[i][0] + "," + closest[i][0] + "] : " + Calculate.Distance(closest[i][0],closest[i][1]));
		}
	}
}

File: c#/ClosestPoints.cs
using System;

class Point
{
	public int x = 0;
	public int y = 0;

	public Point(int x, int y)
	{
		this.x = x;
		this.y = y;
	}
}

class Calculate
{
	public static double Distance(Point point)
	{
		return Math.Sqrt(Math.Pow(Math.Abs(point.x), 2) + Math.Pow(Math.Abs(point.y), 2));
	}
}

class Solution
{
	public static void Sort(Point[] array)
	{
		int minIndex = 0;

		for(int i = 0; i < array.GetLength(0); i++)
		{
			minIndex = i;

			for(int j = i + 1; j < array.GetLength(0); j++)
			{
				if(Calculate.Distance(array[j]) < Calculate.Distance(array[minIndex])) minIndex = j;
			}

			int a0 = array[i].x;
			int a1 = array[i].y;
			int b0 = array[minIndex].x;
			int b1 = array[minIndex].y;

			array[i].x = b0;
			array[i].y = b1;
			array[minIndex].x = a0;
			array[minIndex].y = a1;
		}
	}

	public static Point[] ClosestPoints(Point[] points, int k)
	{
		if(points.Length == 0) return points;
		Point[] closest = new Point[k];

		Sort(points);

		for(int i = 0; i < k; i++)
		{
			closest[i] = new Point(points[i].x, points[i].y);
		}

		return closest;
	}
}

class LeetCode
{
	static void Main()
	{
		int k = 4;
		int size = 10;
		Point[] points = new Point[size];
		points[0] = new Point(0,0);
		points[1] = new Point(1,1);
		points[2] = new Point(2,2);
		points[3] = new Point(3,3);
		points[4] = new Point(4,4);
		points[5] = new Point(5,5);
		points[6] = new Point(6,6);
		points[7] = new Point(7,7);
		points[8] = new Point(8,8);
		points[9] = new Point(9,9);

		Point[] closest = Solution.ClosestPoints(points, k);

		for(int i = 0; i < k; i++)
		{
			Console.WriteLine("[{0},{1}] : {2}", closest[i].x, closest[i].y, Calculate.Distance(closest[i]));
		}
	}
}
